apps/etterlatte-beregning/src/main/kotlin/beregning/BeregnBarnepensjonService.kt

```diff
@@ -295,4 +313,28 @@ class BeregnBarnepensjonService(
     companion object {
         private const val FASTSATT_TRYGDETID_I_PILOT = 40
     }
+
+    private fun List<Grunnlagsdata<JsonNode>>.toPeriodisertAvdoedeGrunnlag():
+        List<GrunnlagMedPeriode<List<Folkeregisteridentifikator>>> {
+        val doede: MutableList<Folkeregisteridentifikator> = mutableListOf()
+        val resultat: MutableList<GrunnlagMedPeriode<List<Folkeregisteridentifikator>>> = mutableListOf()
+
+        val avdoedeSortert = this.distinctBy { it.hentFoedselsnummer() }
+            .sortedBy { it.hentDoedsdato()!!.verdi!! }
+            .toList()
+        val iterator = avdoedeSortert.listIterator()
+        while (iterator.hasNext()) {
+            val dennePeriode = iterator.next()
+            doede.add(Folkeregisteridentifikator.of(dennePeriode.hentFoedselsnummer()!!.verdi.value))
+
+            val nestePeriode = if (iterator.hasNext()) avdoedeSortert[iterator.nextIndex()] else null
+            val datoFOM = virkAvDoedsfall(dennePeriode.hentDoedsdato()!!.verdi)!!
+            val datoTOM = virkAvDoedsfall(nestePeriode?.hentDoedsdato()?.verdi)?.minusDays(1)
+
+            resultat.add(GrunnlagMedPeriode(doede.toImmutableList(), datoFOM, datoTOM))
+        }
+        return resultat.toImmutableList()
```

Comment:
```suggestion
            resultat.add(GrunnlagMedPeriode(doede.toList(), datoFOM, datoTOM))
        }
        return resultat.toList()
```

siden `List` er immutable by default -- eller kanskje også bare skippe det, siden returtypen `List<GrunnlagMedPeriode<List<Folkeregisteridentifikator>>>` tillater ikke modifisering av det du returnerer